### ISSUE TITLE
com.hazelcast.internal.metrics.impl.MetricsCompressorTest [HZ-2538]

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast#23504]
_master_ (commit 65933d5f0689250b39ad958322ee6fb77baaf936)

Failed on ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-fips/56/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

Almost all tests in `com.hazelcast.internal.metrics.impl.MetricsCompressorTest` failed with the same expcetion.

<details><summary>Stacktrace (for testTwoMetrics_secondWithoutTags as example):</summary>

```
Wanted but not invoked:
defaultMetricDescriptorSupplier.get();
-> at com.hazelcast.internal.metrics.impl.DefaultMetricDescriptorSupplier.get(DefaultMetricDescriptorSupplier.java:32)
Actually, there were zero interactions with this mock.

	at com.hazelcast.internal.metrics.impl.DefaultMetricDescriptorSupplier.get(DefaultMetricDescriptorSupplier.java:32)
	at com.hazelcast.internal.metrics.impl.DefaultMetricDescriptorSupplier.get(DefaultMetricDescriptorSupplier.java:27)
	at com.hazelcast.internal.metrics.impl.MetricsCompressorTest.testTwoMetrics_secondWithoutTags(MetricsCompressorTest.java:221)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:825)

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-IbmJDK8-fips/56/

**Comment by Patras3** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/job/Hazelcast-master-Openj9-11/116/testReport/junit/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8/222/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by sumnerib** [comment]
Failed again on _master_ with ibm-8: https://jenkins.hazelcast.com/job/Hazelcast-master-IbmJDK8-fips/63/testReport/junit/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by Patras3** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/job/Hazelcast-master-Openj9-11/119/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8/224/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by Patras3** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/job/Hazelcast-master-Openj9-11/121/testReport/junit/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by sumnerib** [comment]
Failed again on _master_ with zing-8: https://jenkins.hazelcast.com/job/Hazelcast-master-ZingJDK8/487/testReport/junit/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/when_tooLongWord_then_metricIgnored__discriminatorTag/

**Comment by olukas** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/123/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by Patras3** [comment]
Failed again on _master_ with ibm-8: https://jenkins.hazelcast.com/job/Hazelcast-master-IbmJDK8/228/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/testSingleLongMetric/

**Comment by olukas** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/125/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/126/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/127/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/128/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-fips/84/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with:
- openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/130/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/
- ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8-fips/86/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _master_ with:
- openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-11/131/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/
- ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-IbmJDK8/235/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/
- openj9-17: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Openj9-17/66/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Failed again on _5.3.z_ with openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-Openj9-11/86/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by Patras3** [comment]
Failed again on _5.3.z_ with openj9-11: https://jenkins.hazelcast.com/job/Hazelcast-5.maintenance-Openj9-11/88/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

**Comment by olukas** [comment]
Reopening as issue occurred again on: _5.3.z_ (commit 1db02d34f9ff6767ec2113e9570a691e6e170d2b)
with ibm-8: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-IbmJDK8/89/testReport/com.hazelcast.internal.metrics.impl/MetricsCompressorTest/

@JamesHazelcast could you please back-port the fix to `5.3.z` branch?

<details><summary>Stacktrace:</summary>

```
Wanted but not invoked:
defaultMetricDescriptorSupplier.get();
-> at com.hazelcast.internal.metrics.impl.DefaultMetricDescriptorSupplier.get(DefaultMetricDescriptorSupplier.java:32)
Actually, there were zero interactions with this mock.

	at com.hazelcast.internal.metrics.impl.DefaultMetricDescriptorSupplier.get(DefaultMetricDescriptorSupplier.java:32)
	at com.hazelcast.internal.metrics.impl.DefaultMetricDescriptorSupplier.get(DefaultMetricDescriptorSupplier.java:27)
	at com.hazelcast.internal.metrics.impl.MetricsCompressorTest.testTwoMetrics_fullDifferent(MetricsCompressorTest.java:194)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:277)
	at java.lang.Thread.run(Thread.java:825)

```
</details>



Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-5.maintenance-IbmJDK8/89/

**Comment by JamesHazelcast** [comment]
5.3.z backport completed, closing this issue
